workitem/link/type_repository_blackbox_test.go

```diff
@@ -0,0 +1,318 @@
+		require.True(t, fxt.WorkItemLinkTypes[0].Equal(*typ))
+	})
+	s.T().Run("not found", func(t *testing.T) {
+		// given
+		linkTypeID := uuid.NewV4()
+		// when
+		typ, err := s.typeRepo.Load(s.Ctx, linkTypeID)
+		// then
+		require.Error(t, err)
+		require.Nil(t, typ)
+	})
+}
+
+func (s *typeRepoBlackBoxTest) TestCreate() {
+	resetFn := s.DisableGormCallbacks()
+	defer resetFn()
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.SpaceTemplates(1))
+		id := uuid.NewV4()
+		typ := link.WorkItemLinkType{
+			ID:              id,
+			Name:            id.String(),
+			Description:     ptr.String("description for WILT " + id.String()),
+			ReverseName:     "reverse name",
+			ForwardName:     "forward name",
+			Topology:        link.TopologyTree,
+			LinkCategoryID:  link.SystemWorkItemLinkCategoryUserID,
+			SpaceTemplateID: fxt.SpaceTemplates[0].ID,
+		}
+		// when
+		createdType, err := s.typeRepo.Create(s.Ctx, &typ)
+		// then
+		require.NoError(t, err)
+		require.NotNil(t, createdType)
+		require.Equal(t, id, createdType.ID)
+		require.Equal(t, typ, *createdType)
+		require.True(t, typ.Equal(*createdType))
+		// check that loaded type is equal as well
+		loadedType, err := s.typeRepo.Load(s.Ctx, createdType.ID)
+		require.NoError(t, err)
+		require.Equal(t, typ, *loadedType)
+		require.True(t, typ.Equal(*loadedType))
+	})
+	s.T().Run("unknown topology (bad parameter error)", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.SpaceTemplates(1))
+		id := uuid.NewV4()
+		typ := link.WorkItemLinkType{
+			ID:              id,
+			Name:            id.String(),
+			Description:     ptr.String("description for WILT " + id.String()),
+			ReverseName:     "reverse name",
+			ForwardName:     "forward name",
+			Topology:        link.Topology("foobar"),
+			LinkCategoryID:  link.SystemWorkItemLinkCategoryUserID,
+			SpaceTemplateID: fxt.SpaceTemplates[0].ID,
+		}
+		// when
+		createdType, err := s.typeRepo.Create(s.Ctx, &typ)
+		// then
+		require.Error(t, err)
+		require.IsType(t, errors.BadParameterError{}, errs.Cause(err))
+		require.Nil(t, createdType)
+	})
+	s.T().Run("empty name (bad parameter error)", func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.SpaceTemplates(1))
+		id := uuid.NewV4()
+		typ := link.WorkItemLinkType{
+			ID:              id,
+			Name:            "",
+			Description:     ptr.String("description for WILT " + id.String()),
+			ReverseName:     "reverse name",
+			ForwardName:     "forward name",
+			Topology:        link.Topology("foobar"),
+			LinkCategoryID:  link.SystemWorkItemLinkCategoryUserID,
+			SpaceTemplateID: fxt.SpaceTemplates[0].ID,
+		}
+		// when
+		createdType, err := s.typeRepo.Create(s.Ctx, &typ)
+		// then
+		require.Error(t, err)
+		require.IsType(t, errors.BadParameterError{}, errs.Cause(err))
+		require.Nil(t, createdType)
+	})
+	s.T().Run("unique name violation (internal error)", func(t *testing.T) {
```

Comment:
can we add one more test for the uniqueness of parent-child type

Reply:
@pranavgore09 what do you mean? If someone wants to add his own parent child type then she or he can do that but it will not have any meaning to the system.